pkgs/games/mindustry/default.nix

```diff
@@ -93,20 +106,84 @@ let
   '';
   installClient = ''
     install -Dm644 desktop/build/libs/Mindustry.jar $out/share/mindustry.jar
+    install -Dm644 core/assets/icons/icon_64.png $out/share/icons/hicolor/64x64/apps/mindustry.png
+    install -Dm644 ${desktopItem}/share/applications/Mindustry.desktop $out/share/applications/Mindustry.desktop
     mkdir -p $out/bin
+
     makeWrapper ${jre}/bin/java $out/bin/mindustry \
       ${stdenv.lib.optionalString stdenv.isLinux "--prefix LD_LIBRARY_PATH : ${alsaLib}/lib"} \
+      ${stdenv.lib.optionalString stdenv.isLinux "--prefix LD_LIBRARY_PATH : ${glew.out}/lib"} \
+      ${stdenv.lib.optionalString stdenv.isLinux "--prefix LD_LIBRARY_PATH : ${SDL2}/lib"} \
       --add-flags "-jar $out/share/mindustry.jar"
-    install -Dm644 core/assets/icons/icon_64.png $out/share/icons/hicolor/64x64/apps/mindustry.png
-    install -Dm644 ${desktopItem}/share/applications/Mindustry.desktop $out/share/applications/Mindustry.desktop
+  '' + stdenv.lib.optionalString stdenv.isLinux ''
+
+    makeWrapper ${jre}/bin/java $out/bin/mindustry-wayland \
+      --prefix LD_LIBRARY_PATH : ${alsaLib}/lib \
+      --prefix LD_LIBRARY_PATH : ${glew-egl.out}/lib \
+      --prefix LD_LIBRARY_PATH : ${SDL2}/lib \
+      --add-flags "-jar $out/share/mindustry.jar"
+
+    pushd ${arc-natives}
+    zip $out/share/mindustry.jar libsdl-arc64.so
+    popd
   '';
+    #pushd ${SDL2}/lib
+    #zip $out/share/mindustry.jar libSDL2.so
+    #popd
   installServer = ''
     install -Dm644 server/build/libs/server-release.jar $out/share/mindustry-server.jar
     mkdir -p $out/bin
     makeWrapper ${jre}/bin/java $out/bin/mindustry-server \
       --add-flags "-jar $out/share/mindustry-server.jar"
   '';
 
+  glew-egl = glew.overrideAttrs (oldAttrs: {
+    pname = "glew-egl";
+    makeFlags = [ "SYSTEM=linux-egl" ];
```

Comment:
Maybe this line should be:
```nix
makeFlags = oldAttrs.makeFlags ++ [ "SYSTEM=linux-egl" ];
```
This way, we pick up other `makeFlags` from the underlying derivation in the future. I know that there is already a `"SYSTEM=...` line in there, but as far as I know `make` will override those flags, so it could work. But please check for yourself, if this change makes sense.